test/Microsoft.AspNetCore.Server.KestrelTests/FrameResponseHeadersTests.cs

```diff
@@ -14,73 +14,6 @@ namespace Microsoft.AspNetCore.Server.KestrelTests
     public class FrameResponseHeadersTests
     {
         [Theory]
-        [InlineData(true)]
-        [InlineData(false)]
-        public void InitialDictionaryContainsServerAndDate(bool addServerHeader)
-        {
-            var serverOptions = new KestrelServerOptions { AddServerHeader = addServerHeader };
-
-            var connectionContext = new ConnectionContext
-            {
-                DateHeaderValueManager = new DateHeaderValueManager(),
-                ServerAddress = ServerAddress.FromUrl("http://localhost:5000"),
-                ServerOptions = serverOptions,
-            };
-            var frame = new Frame<object>(application: null, context: connectionContext);
-            frame.InitializeHeaders();
-
-            IDictionary<string, StringValues> headers = frame.ResponseHeaders;
-
-            if (addServerHeader)
-            {
-                Assert.Equal(2, headers.Count);
-
-                StringValues serverHeader;
-                Assert.True(headers.TryGetValue("Server", out serverHeader));
-                Assert.Equal(1, serverHeader.Count);
-                Assert.Equal("Kestrel", serverHeader[0]);
-            }
-            else
-            {
-                Assert.Equal(1, headers.Count);
-
-                StringValues serverHeader;
-                Assert.False(headers.TryGetValue("Server", out serverHeader));
-            }
-
-            StringValues dateHeader;
-            DateTime date;
-            Assert.True(headers.TryGetValue("Date", out dateHeader));
-            Assert.Equal(1, dateHeader.Count);
-            Assert.True(DateTime.TryParse(dateHeader[0], out date));
-            Assert.True(DateTime.Now - date <= TimeSpan.FromMinutes(1));
-
-            Assert.False(headers.IsReadOnly);
-        }
-
-        [Fact]
-        public void InitialEntriesCanBeCleared()
```

Comment:
I'd rather have this converted into a test that checks the new initial header count.